github/compliance/milestone.py

```diff
@@ -1,25 +1,51 @@
+import collections.abc
+import dataclasses
 import datetime
 import functools
 import logging
 
 import github3.repos
 
 import delivery.client
-import delivery.model
+import delivery.model as dm
 
 
 logger = logging.getLogger(__name__)
 
 
-def _milestone_title(sprint: delivery.model.Sprint) -> str:
-    return f'sprint-{sprint.name}'
+@dataclasses.dataclass(frozen=True)
+class MilestoneConfiguration:
+    title_callback: collections.abc.Callable[[dm.Sprint], str] = lambda sprint: sprint.name
+    title_prefix: str | None = 'sprint-'
+    title_suffix: str | None = None
+    due_date_callback: collections.abc.Callable[[dm.Sprint], datetime.datetime] \
+        = lambda sprint: sprint.find_sprint_date('release_decision').value
+
+
+def _milestone_title(
+    sprint: dm.Sprint,
+    milestone_cfg: MilestoneConfiguration=None,
+) -> str:
+    if not milestone_cfg:
+        milestone_cfg = MilestoneConfiguration()
+
+    title = milestone_cfg.title_callback(sprint)
+
+    if not (title_prefix := milestone_cfg.title_prefix):
```

Comment:
stylistic nit: might be nicer using `or`, like so:

```
title_prefix = milestone_cfg.title_suffix or ''
...
```

wdyt?

Reply:
I like it, will adjust accordingly.